clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -75,6 +82,8 @@
 
     /* max time in ms for the producer to wait for acknowledgement from server*/
     private final int requestTimeoutMs;
+    
+    private final Time time;
```

Comment:
For all the callers of poll() will pull ```time.milliseconds()``` to get the current timestamp right before this function call except ```Sender.run()``` which will use the pulled timestamp to check buffer expiration as well as setting producer requests' create / send time before calling ```poll```. So if we remove this timestamp from ```KafkaClient``` interfaces we will introduce one more system call for each sender iteration. Given this I think this is OK to keep two longs in the parameters.